x-pack/platform/plugins/shared/alerting/server/application/rule/methods/get_rule_ids_with_gaps/get_rule_ids_with_gaps.ts

```diff
@@ -44,7 +65,7 @@ export async function getRuleIdsWithGaps(
       throw error;
     }
 
-    const { start, end, statuses } = params;
+    const { start, end, statuses, highestPriorityGapFillStatuses } = params;
```

Comment:
nit:

```suggestion
    const { start, end, statuses, highestPriorityGapFillStatuses = [] } = params;
```

will help to simplify

```ts
if (highestPriorityGapFillStatuses?.length ?? 0 > 0) {
```

to

```ts
if (highestPriorityGapFillStatuses.length > 0) {
```